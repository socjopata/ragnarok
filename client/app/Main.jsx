import React, { Component } from "react";
import PropTypes from "prop-types";

export default class Main extends Component {
  static propTypes = {
    testTextPropTypes: PropTypes.string.isRequired,
  }

  static defaultProps = {
    testTextPropTypes: "Kreator Postaci Ragnarok™® - Radosław Jędryszczak™®",
  }

  add = (a, b) => a + b

  render() {
    return (
      <div>
        <h3 className="test-class">
          {this.props.testTextPropTypes}
        </h3>
      </div>
    );
  }
}
