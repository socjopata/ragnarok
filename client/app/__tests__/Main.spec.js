// Docs: https://reactjs.org/docs/test-utils.html
// https://reactjs.org/docs/test-renderer.html
// Analyze setup correctly, buildSetup.js is making this magic and easy testing

import Main from "../Main"

describe("<Main />", () => {
  const defaultProps = {
    testTextPropTypes: "Default text for testing"
  }

  const setup = buildSetup(Main, defaultProps)

  test("render", () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot()
    // For now React 16 can return array of children from return, not a single jsx statement. That's why array is expected
    expect(wrapper.root.findByProps({ className: "test-class" }).children).toEqual(["Default text for testing"])
  })

  describe("add", () => {
    test("should add 2 numbers", () => {
      const { wrapper } = setup();
      const wrapperInstance = wrapper.getInstance()

      expect(wrapperInstance.add(1, 2)).toEqual(3)
    })
  })
})
