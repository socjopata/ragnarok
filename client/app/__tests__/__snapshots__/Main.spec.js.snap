// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> render 1`] = `
<div>
  <h3
    className="test-class"
  >
    Default text for testing
  </h3>
</div>
`;
